### ISSUE TITLE
Incorrect error reporting from `prefer-global-this` rule

### ISSUE THREAD

**Issue by charles4221** [sindresorhus/eslint-plugin-unicorn#2489]
I just upgraded my version of the plugin to `56.0.0` and started getting lint errors for the `prefer-global-this` rule on code like this:

```ts
export function detectPrefersDarkMode() {
  return (
    typeof window !== 'undefined' &&
    window.matchMedia('(prefers-color-scheme: dark)').matches
  );
}
```

Shouldn't this type of usage be excluded from this rule since `globalThis.matchMedia` will crash with a `TypeError: globalThis.matchMedia is not a function`?

The above example is also a common use case in apps that run on both server and client, to check that the browser environment exists before running browser-specific code. `globalThis` isn't going to work for those use cases.

**Comment by sindresorhus** [comment]
This is a faulty check. You can no longer make this assumption, regardless of this rule. The correct way is to test the individual functions. I think this rule is doing you a favor by leading you to a better pattern.

```js
export function detectPrefersDarkMode() {
  return globalThis.matchMedia?.('(prefers-color-scheme: dark)').matches ?? false
}
```

**Comment by fregante** [comment]
A couple of notes:

- `window === globalThis` in the browser, so this definitely works there
- the only scenario where this piece of code would fail is something like JSDom, where `globalThis.window` _does not_ equal to `globalThis`

So I suggest setting `globalThis.matchMedia` just like you set `globalThis.window`. Kind of annoying I understand, but there's a reason why jsdom specifically warns against that pattern in [their readme](https://github.com/jsdom/jsdom?tab=readme-ov-file#executing-scripts):

> We strongly advise against trying to "execute scripts" by mashing together the jsdom and Node global environments (e.g. by doing global.window = dom.window), and then executing scripts or test code inside the Node global environment. Instead, you should treat jsdom like you would a browser, and run all scripts and tests that need access to a DOM inside the jsdom environment, using window.eval or runScripts: "dangerously". This might require, for example, creating a browserify bundle to execute as a <script> element—just like you would in a browser.

Folding into:

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2488

as a documentation/teaching issue.

**Comment by fregante** [comment]
Duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2468
